WooCommerce/WooCommerceTests/ViewRelated/Products/Filters/FilterProductListViewModelTests.swift

```diff
@@ -0,0 +1,26 @@
+import XCTest
+@testable import WooCommerce
+
+final class FilterProductListViewModelTests: XCTestCase {
+    func testCriteriaWithDefaultFilters() {
+        let filters = FilterProductListViewModel.Filters()
+        let viewModel = FilterProductListViewModel(filters: filters)
+        let expectedCriteria = FilterProductListViewModel.Filters(stockStatus: nil, productStatus: nil, productType: nil, numberOfActiveFilters: 0)
+        XCTAssertEqual(viewModel.criteria, expectedCriteria)
+    }
+
+    func testCriteriaWithNonNilFilters() {
+        let filters = FilterProductListViewModel.Filters(stockStatus: .inStock, productStatus: .draft, productType: .grouped, numberOfActiveFilters: 3)
```

Comment:
⚠️ So, I changed the `numberOfActiveFilters` in this line to `100` and the test still passed! 😁 

It looks like we implement our own `Equatable` here:

https://github.com/woocommerce/woocommerce-ios/blob/eec8ec29b4cfc8e215ecd1550d6f5ee434d07dcd/WooCommerce/Classes/ViewRelated/Products/Filters/FilterProductListViewModel.swift#L124-L130

So, the `numberOfActiveFilters` was **not included**. Was there a reason that we don't include it? I don't remember. 😄 

If we should include it, in case you don't know, the Equatable can now be [generated by the compiler for us](https://useyourloaf.com/blog/how-to-get-equatable-and-hashable-for-free/).

Reply:
ooh great catch and tip! from the post, there's a default one to compare all fields and we implement it manually only if we have a different implementation:

> Now with Swift 4.1 you can remove the boilerplate extension code. You only need to tell the compiler that you want this type to be Equatable by declaring conformance to the protocol

updated in: https://github.com/woocommerce/woocommerce-ios/pull/2328/commits/31bfd56a4f54e9352993dd303d97170feececaae

now it should fail when you change the `numberOfActiveFilters` in this line :)